hbase-server/src/main/java/org/apache/hadoop/hbase/master/procedure/RestoreSnapshotProcedure.java

```diff
@@ -351,6 +352,12 @@ private void prepareRestore(final MasterProcedureEnv env) throws IOException {
         mfs.getFileSystem(),
         SnapshotDescriptionUtils.getCompletedSnapshotDir(snapshot, mfs.getRootDir()),
         snapshot);
+
+      // Verify snapshot validity
+      SnapshotReferenceUtil
```

Comment:
This is currently in a block for only checking non-system tables. Needs to happen for all tables.

Also should probably be the first thing we do in prepareRestore, since it used to proceed the step.

Please update the status monitor about doing the verification as well.